File: Producto/VeciZervi-BackEnd/backend/src/main/java/com/vecizervi/backend/model/Postulacion.java
package com.vecizervi.backend.model;

import jakarta.persistence.*;
import org.hibernate.annotations.CreationTimestamp;
import java.time.LocalDateTime;

@Entity
@Table(name = "postulaciones")
public class Postulacion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_postulacion")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "id_trabajo", nullable = false)
    private Trabajo trabajo;

    @ManyToOne
    @JoinColumn(name = "id_trabajador", nullable = false)
    private Usuario trabajador;

    @Column(name = "mensaje_presentacion", columnDefinition = "TEXT")
    private String mensajePresentacion;

    @CreationTimestamp
    @Column(name = "fecha_postulacion", updatable = false)
    private LocalDateTime fechaPostulacion;

    public Postulacion() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Trabajo getTrabajo() { return trabajo; }
    public void setTrabajo(Trabajo trabajo) { this.trabajo = trabajo; }
    public Usuario getTrabajador() { return trabajador; }
    public void setTrabajador(Usuario trabajador) { this.trabajador = trabajador; }
    public String getMensajePresentacion() { return mensajePresentacion; }
    public void setMensajePresentacion(String mensajePresentacion) { this.mensajePresentacion = mensajePresentacion; }
    public LocalDateTime getFechaPostulacion() { return fechaPostulacion; }
}

File: Producto/Android Studio/VeciZerviGub/app/src/main/java/com/example/vecizervi/data/models/Postulacion.kt
package com.example.vecizervi.data.models

import com.google.gson.annotations.SerializedName

data class Postulacion(
    @SerializedName("id_postulacion") val idPostulacion: Long? = null,
    @SerializedName("id_trabajo") val idTrabajo: Long,
    @SerializedName("id_trabajador") val idTrabajador: Long,
    @SerializedName("mensaje_presentacion") val mensajePresentacion: String = "",
    @SerializedName("fecha_postulacion") val fechaPostulacion: String = ""
)
